### PR TITLE
Refactor github actions

### PR DESCRIPTION
**Pull request by LucioFranco**

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -7,161 +7,125 @@ on:
...
+    - uses: actions/checkout@master
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: ${{ matrix.rust }}
+        profile: minimal
+    - name: Check
+      uses: actions-rs/cargo@v1
+      with:
+        command: check
+        args: --all --bins --examples --tests --benches --all-features
```
Comment (jonhoo) [inline .github/workflows/CI.yml:26]: Why does this explicitly enumerate instead of using, say, `--all-targets`?

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -7,161 +7,125 @@ on:
...
-
     strategy:
       matrix:
-        crate:
-          - tower
-          - tower-layer
-          - tower-service
-          - tower-test
-
+        rust: [stable, beta, nightly, 1.39.0]
```
Comment (jonhoo) [inline .github/workflows/CI.yml:80]: Is there a way for us to allow failures on nightly?

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -7,161 +7,125 @@ on:
...
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+        components: rustfmt
+        profile: minimal
+    - name: rustfmt
+      uses: actions-rs/cargo@v1
+      with:
+        command: fmt
+        args: --all -- --check
```
Comment (jonhoo) [inline .github/workflows/CI.yml:108]: Should we also run a clippy check?

File: .github/workflows/CI.yml [hunk 4]
```diff
@@ -7,161 +7,125 @@ on:
...
-#          toolchain: nightly
-#          override: true
-#          components: clippy
-#      - name: Run clippy
-#        uses: actions-rs/clippy-check@v1
-#        with:
-#          token: ${{ secrets.GITHUB_TOKEN }}
-#          args: --all-targets --all-features
+    - uses: actions/checkout@v1
+    - uses: EmbarkStudios/cargo-deny-action@v0
```
Comment (jonhoo) [inline .github/workflows/CI.yml:131]: newline

File: tower-service/Cargo.toml [hunk 5]
```diff
@@ -24,4 +24,4 @@ edition = "2018"
 [dependencies]
 
 [dev-dependencies]
-http = "0.1"
+http = "0.2"
```
Comment (jonhoo) [inline tower-service/Cargo.toml:27]: This seems unrelated?

File: tower/Cargo.toml [hunk 6]
```diff
@@ -60,7 +60,7 @@ tokio = { version = "0.2", optional = true }
 [dev-dependencies]
 futures-util = { version = "0.3", default-features = false, features = ["alloc", "async-await"] }
 hdrhistogram = "6.0"
-quickcheck = { version = "0.6", default-features = false }
+quickcheck = { version = "0.9", default-features = false }
```
Comment (jonhoo) [inline tower/Cargo.toml:63]: This seems unrelated?

File: deny.toml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+[licenses]
+unlicensed = "deny"
+# We want really high confidence when inferring licenses from text
+confidence-threshold = 0.92
+allow = [
```
Comment (jonhoo) [inline deny.toml:5]: Why this particular list?

File: tower-service/Cargo.toml (hunk 5 above)
Comment (LucioFranco) [reply tower-service/Cargo.toml:27]: Cargo deny was complaining, duplicate bytes versions.

File: tower/Cargo.toml (hunk 6 above)
Comment (LucioFranco) [reply tower/Cargo.toml:63]: Same with the above, duplicate versions of rand.

File: deny.toml (hunk 7 above)
Comment (LucioFranco) [reply deny.toml:5]: I copied mostly from tonic but should include a list of licenses we accept.

Comment (LucioFranco) [reply deny.toml:5]: This can be expanded though.

Comment (jonhoo) [reply deny.toml:5]: If it helps, here is flurry's [`deny.toml`](https://github.com/jonhoo/flurry/blob/master/deny.toml).

File: .github/workflows/CI.yml (hunk 3 above)
Comment (LucioFranco) [reply .github/workflows/CI.yml:108]: There is but we have a lot of warnings so I am going to try and fix that in another PR.

File: .github/workflows/CI.yml (hunk 2 above)
Comment (LucioFranco) [reply .github/workflows/CI.yml:80]: uhhhhhhh not sure!

File: .github/workflows/CI.yml (hunk 1 above)
Comment (LucioFranco) [reply .github/workflows/CI.yml:26]: not totally sure will replace that

File: .github/workflows/CI.yml (hunk 2 above)
Comment (LucioFranco) [reply .github/workflows/CI.yml:80]: doesn't seem possible
